src/components/patch/PatchContainer.tsx

```diff
@@ -39,7 +41,7 @@ export class PatchContainer extends React.Component<Props, State> {
       <Grid className="patch-container" container={true} spacing={24}>
         {this.state.visiblePatches.map(patchObj => (
           <Grid item={true} xs={12} key={patchObj.Version.id}>
-            <Patch Patch={patchObj} />
+            <Patch Patch={patchObj} updateOpenPatches={this.updateOpenPatches} expanded={this.isExpanded(patchObj)} />
```

Comment:
This check is O(n^2) since indexOf needs to potentially scan the entire patches array. Can you instead track the expanded patches by whatever unique key you're using to repeat them?

Reply:
Good call, I totally forgot indexOf would be O(n^2). I'm now representing those with a dict using the patch ID as a key.